aopy/utils/base.py

```diff
@@ -795,4 +795,31 @@ def multiply_mat_batch(data, mat, save_path, scale = 1, max_memory_gb = 1., dtyp
     for ibatch in range(Nbatches):
         tmp = mat @ (scale*data[ibatch*batch_size:(ibatch+1)*batch_size, :].T)
         x[ibatch*batch_size:(ibatch+1)*batch_size, :] = (tmp.T).astype(dtype)
-        x.flush()
\ No newline at end of file
+        x.flush()
+
+def generate_poisson_timestamps(mu, max_time, min_time=0., refractory_period=0.):
+    """
+    Generate timestamps following a Poisson process with mean time between events mu,
+    with a specified minimum refractory period, and that fall within a specified time window.
```

Comment:
Its probably worth explicitly explaining that the number of timestamps is defined by the input parameters (mu, max_time, min_time)

Reply:
Also, that this will vary every time the function is called unless a specified seed is set.